izpack-installer/src/main/java/com/izforge/izpack/installer/gui/InstallerFrame.java

```diff
@@ -304,6 +305,9 @@ public void buildGUI()
 
         Messages messages = locales.getMessages();
         navigator.updateButtonText(messages);
+        
+        // set the frame title
+        setTitle(getTitle(installdata));
```

Comment:
Needless to say, please test this carefully in both modes, GUI and console, also with unit tests included, to not break anything.